cli/.js/src/main/scala/aqua/run/RunOpts.scala

```diff
@@ -58,32 +63,109 @@ object RunOpts extends Logging {
+                      )
+                    }
+                    .compile
+                    .last
+                    .map(_.map(_.map(v => Some(v))).getOrElse(validNec(None)))
+                } else {
+                  invalidNec(s"Path '${p.toString}' is not a file").pure[F]
+                }
+              }
+            else {
+              invalidNec(s"There is no path '${p.toString}'").pure[F]
+            }
+          }
+      }
+
+  val dataOpt: Opts[js.Dynamic] =
+    Opts.option[String]("data", "Data for aqua function in json", "d").mapValidated { str =>
+      Validated.catchNonFatal {
+        JSON.parse(str)
+      }.leftMap(t => NonEmptyList.one("Data isn't a valid JSON: " + t.getMessage))
+    }
+
+  val funcOpt: Opts[(String, List[ValueModel])] =
     Opts
       .option[String]("func", "Function to call with args", "f")
       .mapValidated { str =>
         CallArrowExpr.funcOnly.parseAll(str) match {
-          case Right(f) =>
-            val expr = f.mapK(spanToId)
-            val hasVars = expr.args.exists {
-              case VarLambda(_, _) => true
-              case _ => false
-            }
-            if (hasVars) {
-              Validated.invalidNel(
-                "Function can have only literal arguments, no variables or constants allowed at the moment"
-              )
-            } else {
-              val args = expr.args.collect { case l @ Literal(_, _) =>
-                LiteralModel(l.value, l.ts)
-              }
+          case Right(exprSpan) =>
+            val expr = exprSpan.mapK(spanToId)
 
-              Validated.validNel((expr.funcName.value, args))
+            val args = expr.args.collect {
+              case Literal(value, ts) =>
+                LiteralModel(value, ts)
+              case VarLambda(name, _) =>
+                VarModel(name.value, BottomType)
             }
-          case Left(err) => Validated.invalid(err.expected.map(_.context.mkString("\n")))
+
+            validNel((expr.funcName.value, args))
+
+          case Left(err) => invalid(err.expected.map(_.context.mkString("\n")))
         }
       }
 
+  // checks if data is presented if there is non-literals in function arguments
+  // creates services to add this data into a call
+  def checkDataGetServices(
+    args: List[ValueModel],
+    data: Option[js.Dynamic]
+  ): ValidatedNec[String, Map[String, GetterBuilder]] = {
+    val vars = args.collect { case v @ VarModel(_, _, _) =>
+      v
+    // one variable could be used multiple times
+    }.distinctBy(_.name)
+
+    data match {
+      case None if vars.nonEmpty =>
+        invalidNec("Function have non-literals, so, data should be presented")
+      case None =>
+        validNec(Map.empty)
+      case Some(data) =>
+        val services = vars.map { vm =>
+          val arg = data.selectDynamic(vm.name)
+          vm.name -> GetterBuilder.create(vm, arg)
```

Comment:
Let it be `GetDataService` or `GetArgumentService` or `ArgumentGetter` or `ArgumentProvider(vm, arg)`